plane/src/dns/mod.rs

```diff
@@ -124,6 +130,24 @@ impl AcmeDnsServer {
                     });
                 };
 
+                let name = if let Some(zone) = &self.zone {
+                    if let Some(name) = name.strip_suffix(zone) {
+                        name.strip_suffix('.').unwrap_or(name)
```

Comment:
maybe put this strip_suffix bit in the parse implementation for cluste name or something? it's a little error prone to have to remember to do this in every code path.